pkg/controller/controller.go

```diff
@@ -150,38 +153,36 @@ func (c *Controller) init() {
 	c.kubernetesFactory.Core().V1().Endpoints().Informer().AddEventHandler(c.handler)
 	c.kubernetesFactory.Core().V1().Pods().Informer().AddEventHandler(c.handler)
 
-	// Create the base listers
-	c.PodLister = c.kubernetesFactory.Core().V1().Pods().Lister()
-	c.EndpointsLister = c.kubernetesFactory.Core().V1().Endpoints().Lister()
-
 	c.deployLog = deploylog.NewDeployLog(c.log, 1000)
 	c.api = api.NewAPI(c.log, c.apiPort, c.apiHost, &c.lastConfiguration, c.deployLog, c.PodLister, c.meshNamespace)
 
-	if c.smiEnabled {
-		// Create new SharedInformerFactories, and register the event handler to informers.
-		c.accessFactory = accessInformer.NewSharedInformerFactoryWithOptions(c.clients.GetAccessClient(), k8s.ResyncPeriod)
-		c.accessFactory.Access().V1alpha1().TrafficTargets().Informer().AddEventHandler(c.handler)
-
-		c.specsFactory = specsInformer.NewSharedInformerFactoryWithOptions(c.clients.GetSpecsClient(), k8s.ResyncPeriod)
-		c.specsFactory.Specs().V1alpha1().HTTPRouteGroups().Informer().AddEventHandler(c.handler)
-		c.specsFactory.Specs().V1alpha1().TCPRoutes().Informer().AddEventHandler(c.handler)
-
-		c.splitFactory = splitInformer.NewSharedInformerFactoryWithOptions(c.clients.GetSplitClient(), k8s.ResyncPeriod)
-		c.splitFactory.Split().V1alpha2().TrafficSplits().Informer().AddEventHandler(c.handler)
-
-		// Create the SMI listers
-		c.TrafficTargetLister = c.accessFactory.Access().V1alpha1().TrafficTargets().Lister()
-		c.HTTPRouteGroupLister = c.specsFactory.Specs().V1alpha1().HTTPRouteGroups().Lister()
-		c.TCPRouteLister = c.specsFactory.Specs().V1alpha1().TCPRoutes().Lister()
-		c.TrafficSplitLister = c.splitFactory.Split().V1alpha2().TrafficSplits().Lister()
-
-		c.provider = smi.New(c.log, c.defaultMode, c.tcpStateTable, c.ignored, c.ServiceLister, c.EndpointsLister, c.PodLister, c.TrafficTargetLister, c.HTTPRouteGroupLister, c.TCPRouteLister, c.TrafficSplitLister, c.minHTTPPort, c.maxHTTPPort)
-
-		return
-	}
-
-	// If SMI is not configured, use the kubernetes provider.
-	c.provider = kubernetes.New(c.log, c.defaultMode, c.tcpStateTable, c.ignored, c.ServiceLister, c.EndpointsLister, c.minHTTPPort, c.maxHTTPPort)
+	// Create new SharedInformerFactories, and register the event handler to informers.
```

Comment:
Since we are changing which informers are being used at any time, we need to update the prepare check to ensure informers start properly.

Reply:
Oh :fearful: We completely forgot about this.